test/core.jl

```diff
@@ -4822,6 +4822,17 @@ end
 @test f14893() == 14893
 @test M14893.f14893() == 14893
 
+# issue #18725
+@test_nowarn begin
```

Comment:
Somehow, it seems that the "Method definition overwritten" error is being printed at some other stage of code evaluation, so that it occurs before the `redirect_stderr` is executed.